src/plugins/data/server/search/strategies/common/async_utils.ts

```diff
@@ -10,29 +10,35 @@ import {
   AsyncSearchSubmitRequest,
   AsyncSearchGetRequest,
 } from '@elastic/elasticsearch/lib/api/types';
-import { SearchSessionsConfigSchema } from '../../../../config';
 import { ISearchOptions } from '../../../../common';
+import { SearchConfigSchema } from '../../../../config';
 
 /**
  @internal
  */
 export function getCommonDefaultAsyncSubmitParams(
-  searchSessionsConfig: SearchSessionsConfigSchema | null,
-  options: ISearchOptions
+  config: SearchConfigSchema,
+  options: ISearchOptions,
+  /**
+   * Allows to override some of internal logic (e.g. eql / sql searches don't fully support search sessions yet)
+   */
+  overrides?: {
+    disableSearchSessions?: true;
```

Comment:
Previously it was possible to provide `null` instead of  `searchSessionsConfig` to _force_ options to behave as if search sessions are disabled. We used this for EQL and SQL search strategies which don't fully support search sessions. 


But now, as we provide more general `search` config, instead of requiring consumer code augmenting `config.sessions.enabled` inside the `config: SearchConfigSchema`, I decided to provide this `overrides` param to force this code to behave as if sessions were disabled. I think this is clearer and less error-prone than if consumer code would be augmenting deeply nested property inside the config object.